cluster-autoscaler/expander/factory/expander_factory.go

```diff
@@ -32,10 +33,25 @@ import (
 )
 
 // ExpanderStrategyFromString creates an expander.Strategy according to its name
-func ExpanderStrategyFromString(expanderFlag string, cloudProvider cloudprovider.CloudProvider,
+func ExpanderStrategyFromString(expanderFlag string, fallbackExpanderFlag string, grpcExpanderCert string, grpcExpanderURL string, cloudProvider cloudprovider.CloudProvider,
 	autoscalingKubeClients *context.AutoscalingKubeClients, kubeClient kube_client.Interface,
 	configNamespace string) (expander.Strategy, errors.AutoscalerError) {
-	switch expanderFlag {
+
+  // For GRPC expander, get the fallback strategy recursively. Ensure fallback expander is not GRPC so there is no infinite loop.
+  if expanderFlag == expander.GRPCExpanderName {
+    if fallbackExpanderFlag == expander.GRPCExpanderName {
+      return nil, errors.NewAutoscalerError(errors.ConfigurationError, "grpc expander is not a valid fallback for itself.")
+    }
+    fallbackExpander, err := ExpanderStrategyFromString(fallbackExpanderFlag, "", grpcExpanderCert, grpcExpanderURL, cloudProvider,
```

Comment:
I also didn't love doing this recursively, but I couldn't think of a more eloquent way to do it

Reply:
seems fine to me.